libr/include/r_types_overflow.h

```diff
@@ -0,0 +1,53 @@
+#ifndef R_TYPES_OVERFLOW_H
+#define R_TYPES_OVERFLOW_H
+
+// TODO: Use CLANG/GCC builtins if available: __builtin_mul_overflow
+
+// ADD
+// if ((x > 0) && (a > INT_MAX - x)) /* `a + x` would overflow */;
+// if ((x < 0) && (a < INT_MIN - x)) /* `a + x` would underflow */;
+#define SZT_ADD_OVFCHK(x,y) ((SIZE_MAX - (x)) < (y))
+#define SSZT_ADD_OVFCHK(a,x) ((((x) > 0) && ((a) > SSIZE_MAX - (x))) || (((x) < 0) && (a) < SSIZE_MIN - (x)))
+#define UT64_ADD_OVFCHK(x,y) ((UT64_MAX - (x)) < (y))
+#define ST64_ADD_OVFCHK(a,x) ((((x) > 0) && ((a) > ST64_MAX - (x))) || (((x) < 0) && (a) < ST64_MIN - (x)))
+#define UT32_ADD_OVFCHK(x,y) ((UT32_MAX - (x)) < (y))
+#define ST32_ADD_OVFCHK(a,x) ((((x) > 0) && ((a) > ST32_MAX - (x))) || (((x) < 0) && (a) < ST32_MIN - (x)))
+#define UT16_ADD_OVFCHK(x,y) ((UT16_MAX - (x)) < (y))
+#define ST16_ADD_OVFCHK(a,b) ( \
```

Comment:
just for consistency put even this define on a single line. it should be as long as the other ones, more or less